pkg/executor/sortexec/parallel_sort_spill_helper.go

```diff
@@ -108,6 +108,12 @@ func (p *parallelSortSpillHelper) spill() (err error) {
 		}
 	}()
 
+	p.setInSpilling()
```

Comment:
Why move these code?

Reply:
> Why move these code?

Before this pr, we check signal with `Consume` function, this may lead to the trigger of the spill and cause dead lock if we set spill flag too early. After this pr, there is no dead lock possibility when spill flag is set too early.